src/core/component.js

```diff
@@ -200,20 +200,27 @@ Component.prototype = {
    * @param {string} attrValue - HTML attribute value.
    *        If undefined, use the cached attribute value and continue updating properties.
    */
-  updateProperties: function (attrValue) {
+  updateProperties: function (attrValue, clobber) {
     var el = this.el;
     var isSinglePropSchema = isSingleProp(this.schema);
     var oldData = extendProperties({}, this.data, isSinglePropSchema);
+    var skipTypeChecking = typeof this.previousAttrValue === 'object' && attrValue === this.previousAttrValue;
+    var currentAttrValue = !clobber && this.attrValue;
+    this.previousAttrValue = attrValue;
```

Comment:
We can define this in the constructor as well `this.previousAttrValue = undefined`